### PR TITLE
[ADD] Option for Odoo 18.0

### PR DESCRIPTION
**Pull request by josep-tecnativa**
CC @Tecnativa TT51095
- Added Odoo 18.0 option
- Added PG 16 option 
- Updated pylint version
- Fixed and temporally adjusted some tests for Odoo 18.0 because some modules are not yet migrated
- Updated oca-addons-repo-template

### CODE REVIEW

**Comment by josep-tecnativa** [comment]
I think this is ready to go. Could you review @pedrobaeza ?

File: .pre-commit-config.yaml.jinja [hunk 1]
```diff
@@ -10,7 +10,7 @@
 {%- elif odoo_version < 17 %}
   {%- set proj_rev.pylint_odoo = "v8.0.20" %}
 {%- else %}
-  {%- set proj_rev.pylint_odoo = "v9.0.4" %}
+  {%- set proj_rev.pylint_odoo = "v9.1.3" %}
```
Comment (pedrobaeza) [inline .pre-commit-config.yaml.jinja:13]: You should put this version only for 18, not for 17. Please change also the ruff version to be inline with OCA.

Comment (josep-tecnativa) [reply .pre-commit-config.yaml.jinja:15]: Okay, I have already defined the appropriate version of pylint for v17 and v18-
AFAIK OCA is still using `ruff v0.1.3`

Comment (josep-tecnativa) [reply .pre-commit-config.yaml.jinja:18]: Thank you!

File: .eslintrc.yml.jinja [hunk 2]
```diff
@@ -1 +1 @@
-{%- include "vendor/oca-addons-repo-template/src/{% if odoo_version > 12 %}.eslintrc.yml{% endif %}" -%}
+{%- include "vendor/oca-addons-repo-template/src/{% if odoo_version > 12 %}.eslintrc.yml{% endif %}.jinja" -%}
```
Comment (pedrobaeza) [inline .eslintrc.yml.jinja:1]:
```suggestion
{%- include "vendor/oca-addons-repo-template/src/{% if odoo_version > 12 %}.eslintrc.yml.jinja{% endif %}" -%}
```

File: tests/test_tasks_downstream.py [hunk 3]
```diff
@@ -378,25 +381,32 @@ def test_test_tasks(
...
-                assert _install_status("account_invoice_refund_link") == "uninstalled"
-                stdout = invoke("test", "--private", "--extra", retcode=None)
-                # Ensure "test_module" and "account_invoice_refund_link" were installed
-                assert _install_status("test_module") == "installed"
-                assert _install_status("account_invoice_refund_link") == "installed"
-                _tests_ran(
-                    stdout, supported_odoo_version, "account_invoice_refund_link"
-                )
+                if supported_odoo_version < 18.0:
+                    # TODO: Remove this condition once 'account_invoice_refund_link' is migrated to Odoo 18.0
```
Comment (pedrobaeza) [inline tests/test_tasks_downstream.py:385]:
```suggestion
                    # TODO: Put 19.0 once 'account_invoice_refund_link' is migrated to Odoo 18.0
```

File: tests/test_tasks_downstream.py [hunk 4]
```diff
@@ -378,25 +381,32 @@ def test_test_tasks(
...
+                    # Ensure "test_module" and "account_invoice_refund_link" were installed
+                    assert _install_status("test_module") == "installed"
+                    assert _install_status("account_invoice_refund_link") == "installed"
+                    _tests_ran(
+                        stdout, supported_odoo_version, "account_invoice_refund_link"
+                    )
             # Test --test-tags
-            if supported_odoo_version >= 12:
+            if supported_odoo_version >= 12 and supported_odoo_version < 18.0:
+                # TODO: Remove this condition once 'account_invoice_refund_link' is migrated to Odoo 18.0
```
Comment (pedrobaeza) [inline tests/test_tasks_downstream.py:408]:
```suggestion
                # TODO: Put 19.0 once 'account_invoice_refund_link' is migrated to Odoo 18.0
```

File: .eslintrc.yml.jinja (hunk 2 above)
Comment (josep-tecnativa) [reply .eslintrc.yml.jinja:1]: The file name is actually `vendor/oca-addons-repo-template/src/{% if odoo_version > 12 %}.eslintrc.yml{% endif %}.jinja`
https://github.com/OCA/oca-addons-repo-template/blob/master/src/%7B%25%20if%20odoo_version%20%3E%2012%20%25%7D.eslintrc.yml%7B%25%20endif%20%25%7D.jinja

Comment (pedrobaeza) [reply .eslintrc.yml.jinja:1]: Ah, sorry, I misunderstood it! :facepalm:

**Pull request merged**